src/app/dataset-view/additional-components/data-component/data-component.ts

```diff
@@ -27,14 +27,17 @@ import { sqlEditorOptions } from "src/app/dataset-block/metadata-block/component
 })
 export class DataComponent extends BaseComponent implements OnInit {
     @Input() public datasetBasics?: DatasetBasicsFragment;
-    @Output() public runSQLRequestEmit = new EventEmitter<string>();
+    @Output() public runSQLRequestEmit = new EventEmitter<DatasetRequestBySql>();
     public sqlEditorOptions = sqlEditorOptions;
     public savedQueries = DataTabValues.savedQueries;
     public sqlRequestCode = `select\n  *\nfrom `;
 
     public sqlErrorMarker: MaybeNull<string> = null;
     public currentSchema: MaybeNull<DatasetSchema> = null;
     public currentData: DataRow[] = [];
+    public defaultLimit = AppValues.SQL_QUERY_LIMIT;
+    public queryLimits: number[] = [AppValues.SQL_QUERY_LIMIT, 100, 200, 500];
```

Comment:
I assume `defaultLimit` and `queryLimit` are constants.

Minor: it's preferrable to mark constants with `readonly` attribute and name in `UPPER_CASE`.
Note, this is not entirely cleaned over the system yet.
I.e., this page has some old violations too, `sqlEditorOptions`

Reply:
This functionality is outdated